src/ipfs/uploadIPFSContent.ts

```diff
@@ -0,0 +1,140 @@
+import axios from 'axios';
+
+import { IPFSContentTypes, ContentType, IPFSContentAndType, IPFSUploadServiceResponse } from '../types/ipfs';
+
+const uploadToIPFS = async (ipfsApiUrl: string | undefined, data: IPFSContentTypes) => {
+  if (!ipfsApiUrl) {
+    throw new Error('IPFS_GATEWAY_URL is not set');
+  }
+
+  // POST data to BE service
+  const ipfsHash = await axios
+    .post<IPFSUploadServiceResponse>(`${ipfsApiUrl}?pin=true`, data)
```

Comment:
The caller has to either pass or set `IPFS_GATEWAY_URL` to the full IPFS URL with the upload endpoint.

Our new BE IPFS upload endpoint is `v2/ipfs`. so they will have to pass or set `<API_URL>/v2/ipfs` as the ipfsApiUrl.

This was also we give the caller the flexibility to change to a different IPFS upload URL if they choose to.